src/ansi-c/expr2c.cpp

```diff
@@ -580,14 +617,18 @@ std::string expr2ct::convert_rec(
 
     const typet &return_type=code_type.return_type();
 
-    // return type may be a function pointer or array
-    const typet *non_ptr_type=&ns.follow(return_type);
-    while(non_ptr_type->id()==ID_pointer)
-      non_ptr_type=&(ns.follow(non_ptr_type->subtype()));
+    if(ns)
+    {
+      // return type may be a function pointer or array
+      const typet *non_ptr_type = &ns->get().follow(return_type);
```

Comment:
Perhaps we could wrap the common ones in methods that return optional variants of what was passed it so can push show these checks?